src/wutil.h

```diff
@@ -145,10 +145,10 @@ struct file_id_t {
     dev_t device{static_cast<dev_t>(-1LL)};
     ino_t inode{static_cast<ino_t>(-1LL)};
     uint64_t size{static_cast<uint64_t>(-1LL)};
-    time_t change_seconds{-1};
-    long change_nanoseconds{-1};
-    time_t mod_seconds{-1};
-    long mod_nanoseconds{-1};
+    time_t change_seconds{std::numeric_limits<time_t>::min()};
+    unsigned long change_nanoseconds{std::numeric_limits<unsigned long>::min()};
+    time_t mod_seconds{std::numeric_limits<time_t>::min()};
+    unsigned long mod_nanoseconds{std::numeric_limits<unsigned long>::min()};
```

Comment:
> Fix issue with very very old files
> 
> touch -m -t 197001010000.00 epoch
> touch -m -t 190212112045.40 old
> touch -m -t 190212112045.39 oldest
> touch -m -t 203801080314.07 newest
> 
> Were behaving differently between eachother when used as a left
> hand side file with -nt and then some non-existant file on the
> right hand side.

If we check if both files exist before comparing mtimes, this shouldn't be necessary, right?

Anyway, your solution of giving non-existant files the oldest time possible works fine too and is probably more elegant.

Though I think that making mod_nanoseconds unsigned is wrong because `timespec.tv_nsec` is a signed type.
(Also `std::numeric_limits<unsigned long>::min()` is better written as 0, but I think we should change that anyway.)

Reply:
Oh you're write about it timespec.tv_nsec - it's `long` here.